docs/en/user_guides/internlm_dataset.md

```diff
@@ -0,0 +1,208 @@
+## Dataset Format
+
+The InternLM training dataset is already tokenized, and is formatted as follows:
+
+```
+{"tokens": [1, -333, -352, -1621, ..., 103028, 13, 2]}
+{"tokens": [1, -333, -352, -1621, ..., 103028, 13, 2]}
+```
+
+Among them, tokens with negative values are not involved in the calculation of loss during the training process.
+
+## 接口介绍
+
+To train the InternLM-Chat model in XTuner, you need to convert the original data format into the XTuner standard dataset format. The core function for processing the InternLM format dataset is as follows:
+
+```python
+def process(dataset_folder=None,
+            cached_folder=None,
+            split='train',
+            pack_to_max_length=False,
+            max_length=2048,
+            shuffle_before_pack=True,
+            map_num_proc=32):
+    ......
+```
+
+Where:
+
+1. dataset_folder: Indicates the path where the training dataset is located. All files ending with `.bin` in the path will be used as training data. Because the processed data will be cached, **this field needs to be provided only the first time when processing the original data**.
+2. cached_folder: Indicates that the processed data is cached in the cached_folder folder.
+3. split: The dataset read through hf datasets is usually a DatasetDict. The corresponding Dataset is obtained through the split variable. The default value 'train' can generally be used.
+4. pack_to_max_length: Whether to concatenate multiple pieces of data into one piece of data for training.
+5. max_length: Indicates that the data processing process will pack multiple training data into a piece of data with a length of max_length. Only effective when pack_to_max_length=True.
+6. shuffle_before_pack: Whether to shuffle the dataset before packing, generally the default True can be used. Only effective when pack_to_max_length=True.
+7. map_num_proc: Use multi-process for data processing. Depending on the situation, the value of map_num_proc can be increased, and it can be set to 96 on a cluster.
```

Comment:
```suggestion
1. `dataset_folder`: Indicates the path where the training dataset is located. All files ending with `.bin` in the path will be used as training data. Because the processed data will be cached, **this field needs to be provided only the first time when processing the original data**.
2. `cached_folder`: Indicates that the processed data is cached in the cached_folder folder.
3. `split`: The dataset read through hf datasets is usually a DatasetDict. The corresponding Dataset is obtained through the split variable. The default value 'train' can generally be used.
4. `pack_to_max_length`: Whether to concatenate multiple pieces of data into one piece of data for training.
5. `max_length`: Indicates that the data processing process will pack multiple training data into a piece of data with a length of max_length. Only effective when pack_to_max_length=True.
6. `shuffle_before_pack`: Whether to shuffle the dataset before packing, generally the default True can be used. Only effective when pack_to_max_length=True.
7. `map_num_proc`: Use multi-process for data processing. Depending on the situation, the value of map_num_proc can be increased, and it can be set to 96 on a cluster.
```